### DISCUSSION TITLE
Text Quality

### DISCUSSION THREAD

**Discussion by shevernitskiy** [pixijs/pixijs#10941] [Q&A] [answered]
How can i achieve the result presented at figma output? Figma variant looks more thinner and a bit smoother, which looks more pretty. Also, I've noticed that my variant a bit blurry on vertical lines.

Code
```ts
const title = new Text()
title.text = state.title.text;
title.style = {
  fontFamily: state.title.font.face,
  fontSize: state.title.font.size,
  fill: state.title.font.color,
  fontWeight: state.title.font.weight as any,
  lineHeight: state.title.font.line_height,
};
```

Screeshots scaled at 150%

Setting are the same:
Font params: Gilroy 80 700 & Gilroy 50 600
Canvas size: 1280x720

My
![image](https://github.com/user-attachments/assets/680df885-d5c1-4d6d-aa90-650ccda2a6c3)

Figma
![image](https://github.com/user-attachments/assets/e3d365cc-b044-44a7-b1b6-fd7f43a3f997)

**Comment by bigtimebuddy** [comment, marked as answer]
We are somewhat constrained by what the browser can to do. Text rendered with Pixi is using the canvas' fillText method. There are not really options here for customizing the antialiasing, sharpness, etc. 

That said, there are two options you may consider. 

1). Rasterizing a BitmapFont might give you more control with a 3rd party tool (e.g., bmGlyph). You'll have to do a little digging to see what tool options are out there. BitmapFont/BitmapText has decisive performance and memory benefits over Text.

2). Try doubling your Text resolution (i.e., text.resolution = 2). The higher resolution comes with a memory cost but sometime produces smoother results. 

Good luck! Report back if you find a solution that works.

**Reply by shevernitskiy** [reply]
Thanks for rapid answer!
2nd solution works! Result isn't the same, but pretty close. Look good. Thanks you!

**Reply by shevernitskiy** [reply]
found interesting article related to topic
https://wdobbie.com/post/gpu-text-rendering-with-vector-textures/

**Reply by shevernitskiy** [reply]
Also there are some upscaling shaders can be used to solve this issue. Can be implemented as filter.
Here is repo with glsl upscaling shaders https://github.com/bloc97/Anime4K . Unfortunatelly i'm not familiar with shaders language to port it to wgsl and make a filter.
But i think such king of filter will be very usefull.

**Comment by shevernitskiy** [comment]
Ok, i found some more info

> It seems like Figma’s engine is doing font smoothing (antialiasing) on the level of the pixel.
> Some browsers supports (by default) subpixel smoothing, which gives sharper results and slight bolder font.
> 
> Note that switching from subpixel rendering to pixel level anti-aliasing, to make Figma and browser aligned, will create lighter font, which will affect, for example, readability when using light text on dark backgrounds.

I know `font-smooth` for CSS, but don't if  `canvas` `fillText` support it...

**Reply by bigtimebuddy** [reply]
You could try using HTMLText instead which will support CSS rules like that. Give it a try.
